statuspage_connector.py

```diff
@@ -0,0 +1,591 @@
+                    return None
+        except Exception as e:
+            self.debug_print(
+                "Integer validation failed. Error occurred while validating integer value. Error: {}".format(str(e))
+            )
+            if allow_zero:
+                error_text = STATUSPAGE_LIMIT_VALIDATION_ALLOW_ZERO_MSG.format(parameter=key)
+            else:
+                error_text = STATUSPAGE_LIMIT_VALIDATION_MSG.format(parameter=key)
+            action_result.set_status(phantom.APP_ERROR, error_text)
+            return None
+
+        return parameter
+
+    def _handle_test_connectivity(self, param):
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Starting connectivity test")
+
+        # make rest call
+        endpoint = 'pages/{}/incidents'.format(self._page_id)  # pragma: allowlist secret
+
+        ret_val, _ = self._make_rest_call(endpoint, action_result)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Return success
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_get_incident(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        incident_id = param.get('incident_id')
+
+        endpoint = 'pages/{0}/incidents/{1}'.format(self._page_id, incident_id)  # pragma: allowlist secret
+
+        ret_val, response = self._make_rest_call(endpoint, action_result)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Add the response into the data section
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully fetched incident")
+
+    def _handle_list_incidents(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        parameters = dict()
+        query = param.get('query')
+        limit = param.get('limit', 100)
+        page = param.get('page_offset')
+
+        limit = self._validate_integers(action_result, param.get('limit', 100), 'limit')
+        if limit is None:
+            return action_result.get_status(), None
+
+        if page or page == 0:
+            page = self._validate_integers(action_result, page, 'page offset', allow_zero=True)
+            if page is None:
+                return action_result.get_status(), None
+
+        if query:
+            parameters['q'] = query
+        if limit:
+            parameters['limit'] = limit
+        if page:
+            parameters['page'] = page
+
+        endpoint = 'pages/{}/incidents'.format(self._page_id)  # pragma: allowlist secret
```

Comment:
It doesn't look like any sensitive information/words, Is detect_secret check failing here?
If not, we can remove the allowlist comment. (+7 other endpoints have the same comment)